src/extensions/ExtensionManager.as

```diff
@@ -106,10 +107,16 @@ public class ExtensionManager {
 	// Return a command spec array for the given operation or null.
 	public function specForCmd(op:String):Array {
 		for each (var ext:ScratchExtension in extensionDict) {
-			var prefix:String = ext.useScratchPrimitives ? '' : (ext.name + extensionSeparator);
 			for each (var spec:Array in ext.blockSpecs) {
-				if ((spec.length > 2) && ((prefix + spec[2]) == op)) {
-					return [spec[1], spec[0], Specs.extensionsCategory, op, spec.slice(3)];
+				if (spec.length > 2) {
+					var compareOp:String = ext.useScratchPrimitives ?
+							spec[2] : (ext.name + extensionSeparator + spec[2]);
+					var legacyOp:String = ext.useScratchPrimitives ?
+							spec[2] : (ext.name + extensionSeparatorLegacy + spec[2]);
+					if (op == compareOp || op == legacyOp) {
```

Comment:
The comparison against `legacyOp` is redundant but harmless if `ext.useScratchPrimitives` is true. It's important that `legacyOp == compareOp` in that case, though, to make sure that the comparisons are identical.

Suppose, for example, that `legacyOp` was initialized to `null` or `undefined` in the `ext.useScratchPrimitives` case. This function could then return an unexpected spec -- specifically, whatever happens to be the first block spec in the first extension -- when called with a `null` or `undefined` op. This function should never be called with a `null` or `undefined` op, but if we ever have such a bug I don't want to make it harder to track down.